azurelinuxagent/common/protocol/wire.py

```diff
@@ -789,58 +789,50 @@ def update_goal_state(self, forced=False):
 
-                try:
-                    vm_settings, response_headers = get_vm_settings()
-                except ResourceGoneError:
-                    self.update_host_plugin_from_goal_state()
-                    vm_settings, response_headers = get_vm_settings()
-
-                # on error, fetch() logs the original exception and returns None as content
-                if vm_settings is None:
-                    raise Exception("Failed to retrieve the extensions goal state (vmSettings), the error was reported previously in the log.")
-
-                # The response includes an etag if and only if the VM settings change.
-                response_etag = None
-                for h in response_headers:
-                    if h[0].lower() == 'etag':
-                        response_etag = h[1]
-                        break
-
-                # TODO: Remove this log statement when invoking this method on each iteration of the goal state loop (currently it is invoked only on a new goal state)
-                logger.info("Fetched extensions goal state [correlation ID: {0} eTag: {1}]", correlation_id, response_etag)
-
-                # TODO: Remove the if clause when removing the retry loop (the else clause should remain)
-                if response_etag is None and wait_for_new_goal_state:
-                    logger.info("Expected a new goal state; will retry after a short delay...")
-                    time.sleep(5)
-                else:
-                    logger.info("Fetched new extensions goal state [correlation ID: {0} eTag: {1}]", correlation_id, response_etag)
-                    self._extensions_goal_state = ExtensionsGoalState(response_etag, vm_settings)
+            try:
+                vm_settings, response_headers = get_vm_settings()
+            except ResourceGoneError:
+                self.update_host_plugin_from_goal_state()
+                vm_settings, response_headers = get_vm_settings()
+
+            # on error, fetch() logs the original exception and returns None as content
+            if vm_settings is None:
+                raise Exception("Failed to retrieve the extensions goal state (vmSettings), the error was reported previously in the log.")
+
+            # The response includes an etag if and only if the VM settings change.
+            response_etag = None
+            for h in response_headers:
+                if h[0].lower() == 'etag':
+                    response_etag = h[1]
                     break
 
+            # TODO: Remove this log statement when invoking this method on each iteration of the goal state loop (currently it is invoked only on a new goal state)
+            logger.info("Fetched extensions goal state [correlation ID: {0} eTag: {1}]", correlation_id, response_etag)
+
+            logger.info("Fetched new extensions goal state [correlation ID: {0} eTag: {1}]", correlation_id, response_etag)
```

Comment:
Any reason for logging the same data twice? (Line 831 and 833)?

Reply:
831 is no longer needed now that the retry loop is gone, let me remove it